### DISCUSSION TITLE
Select Show right unique items shows filtered directories

### DISCUSSION THREAD

**Discussion by acotty** [WinMerge/winmerge#1337] [Q&A]
If I show differences the filtered directories are not shown as expected.
If I then tick show right unique items the filtered directories are now shown if there are right unique items in the directory tree.

Is there a way of enabling the directory filter with the show right unique items so that the filtered directory tree is not shown when showing  unique items? This could be a bug, but hopefully it's a config item that is not obvious.

**Comment by sdottaka** [comment]
Could you attach specific examples or screenshots?
Also, what version of WinMerge are you using?

**Comment by acotty** [comment]
I am using Winmerge x64 2.16.20.0  (WinMerge-2.16.20-x64-Setup.exe) on Windows 10 21H2 with latest patches from last week.

I have attached the following:
1) screen shot of the issue
2) the filter I am using 
3) Exported config

The comparison is between a CodeBlocks build directory and the Git repo that as you can appreciate does not have the objects or release files (devel<31 or 31_64 etc> directory name depending on the build type 32 V 64 and/or Windows  V Linux V Macos).

I went through the config options again (in the morning) and spotted the "Include unique subfolder contents" causes the output to include the files in the filtered folders, but the filtered directory is still shown, see E:\Temp\2022-05-17 10_37_31-WinMerge - Unticked unique subfolder contents .png for the output when the "Include unique subfolder contents" is unticked.


![2022-05-17 10_27_48-WinMerge_2 16 20 0](https://user-images.githubusercontent.com/13925856/168703397-2f2e7278-2155-417d-b35a-2a09cae33724.png
[CodeBlocks.flt.txt](https://github.com/WinMerge/winmerge/files/8704195/CodeBlocks.flt.txt)
[winmerge_options.ini.txt](https://github.com/WinMerge/winmerge/files/8704202/winmerge_options.ini.txt)
![2022-05-17 10_37_31-WinMerge - Unticked unique subfolder contents ](https://user-images.githubusercontent.com/13925856/168704136-53b8502e-4021-43a4-853b-b49dc06d4b9b.png)
)

**Reply by sdottaka** [reply]
Thank you for uploading the files

Are the folders that were unintentionally displayed as follows?

- .objs31_64
- devel31_64
- output31_64

If so, the attached file filter does not match the above folders, so you need to change it as follows.

`\\objs$` → `\\ objs.*$`
`\\develop$` → `\\develop.*$`
`\\output$` → `\\output.*$`


![image](https://user-images.githubusercontent.com/98126/168802155-c5201543-b9b3-4bb6-bdea-f2451a155dac.png)
